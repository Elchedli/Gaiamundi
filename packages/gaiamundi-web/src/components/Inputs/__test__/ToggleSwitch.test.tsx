import { fireEvent, render } from '@testing-library/react';
import { ToggleSwitch } from '../ToggleSwitch';

describe('ToggleSwitch', () => {
  it('renders correctly', () => {
    const { container } = render(
      <ToggleSwitch label="Test" defaultChecked={false} onChange={jest.fn()} />
    );
    expect(container).toMatchSnapshot();
  });

  it('calls onChange function when switch is clicked', () => {
    const onChange = jest.fn();
    const { getByTestId } = render(
      <ToggleSwitch label="Test" defaultChecked={false} onChange={onChange} />
    );
    fireEvent.click(getByTestId('toggle-switch'));
    expect(onChange).toHaveBeenCalledWith(true);
  });

  it('displays the correct label', () => {
    const label = 'Test Label';
    const { getByText } = render(
      <ToggleSwitch label={label} defaultChecked={false} onChange={jest.fn()} />
    );
    expect(getByText(label)).toBeInTheDocument();
  });

  it('starts with the correct default state and action working on toggle', async () => {
    const defaultChecked = true;
    const { getByTestId } = render(
      <ToggleSwitch
        label="Test"
        defaultChecked={defaultChecked}
        onChange={jest.fn()}
      />
    );
    const toggleSwitch = getByTestId('toggle-switch');
    expect(toggleSwitch).toHaveClass('bg-blue-600');
    fireEvent.click(toggleSwitch);
    expect(toggleSwitch).toHaveClass('bg-gray-200');
  });
});
