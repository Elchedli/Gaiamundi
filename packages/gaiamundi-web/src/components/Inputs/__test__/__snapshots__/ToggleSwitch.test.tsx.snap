// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleSwitch calls onChange function when switch is clicked 1`] = `
<div>
  <div
    class="flex items-center"
  >
    <button
      aria-checked="true"
      aria-labelledby="headlessui-label-:r3:"
      class="bg-blue-600 relative inline-flex h-6 w-11 items-center rounded-full transition-colors focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2"
      data-headlessui-state="checked"
      data-testid="toggle-switch"
      id="headlessui-switch-:r2:"
      role="switch"
      tabindex="0"
      type="button"
    >
      <span
        class="translate-x-6 inline-block h-4 w-4 transform rounded-full bg-white transition-transform"
      />
    </button>
    <label
      class="ml-4 cursor-pointer"
      id="headlessui-label-:r3:"
    >
      Test
    </label>
  </div>
</div>
`;

exports[`ToggleSwitch displays the correct label 1`] = `
<div>
  <div
    class="flex items-center"
  >
    <button
      aria-checked="false"
      aria-labelledby="headlessui-label-:r5:"
      class="bg-gray-200 relative inline-flex h-6 w-11 items-center rounded-full transition-colors focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2"
      data-headlessui-state=""
      data-testid="toggle-switch"
      id="headlessui-switch-:r4:"
      role="switch"
      tabindex="0"
      type="button"
    >
      <span
        class="translate-x-1 inline-block h-4 w-4 transform rounded-full bg-white transition-transform"
      />
    </button>
    <label
      class="ml-4 cursor-pointer"
      id="headlessui-label-:r5:"
    >
      Test Label
    </label>
  </div>
</div>
`;

exports[`ToggleSwitch renders correctly 1`] = `
<div>
  <div
    class="flex items-center"
  >
    <button
      aria-checked="false"
      aria-labelledby="headlessui-label-:r1:"
      class="bg-gray-200 relative inline-flex h-6 w-11 items-center rounded-full transition-colors focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2"
      data-headlessui-state=""
      data-testid="toggle-switch"
      id="headlessui-switch-:r0:"
      role="switch"
      tabindex="0"
      type="button"
    >
      <span
        class="translate-x-1 inline-block h-4 w-4 transform rounded-full bg-white transition-transform"
      />
    </button>
    <label
      class="ml-4 cursor-pointer"
      id="headlessui-label-:r1:"
    >
      Test
    </label>
  </div>
</div>
`;

exports[`ToggleSwitch starts with the correct default state and action working on toggle 1`] = `
<div>
  <div
    class="flex items-center"
  >
    <button
      aria-checked="false"
      aria-labelledby="headlessui-label-:r7:"
      class="bg-gray-200 relative inline-flex h-6 w-11 items-center rounded-full transition-colors focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2"
      data-headlessui-state=""
      data-testid="toggle-switch"
      id="headlessui-switch-:r6:"
      role="switch"
      tabindex="0"
      type="button"
    >
      <span
        class="translate-x-1 inline-block h-4 w-4 transform rounded-full bg-white transition-transform"
      />
    </button>
    <label
      class="ml-4 cursor-pointer"
      id="headlessui-label-:r7:"
    >
      Test
    </label>
  </div>
</div>
`;
